src/tools/perf/lib/ucp_tests.cc

```diff
@@ -90,16 +90,36 @@ class ucp_perf_test_runner {
         }
     }
 
+    void make_blocking_progress() {
+        int count = 0;
+
+        while (ucp_worker_progress(m_perf.ucp.worker)) {
+            count++;
+        }
+
+        if (!count) {
```

Comment:
why needed?

Reply:
If `count` is zero, then worker is drained and it's safe to call worker_wait.

Reply:
if we get to this point, it means last call to progress returned 0